code/modules/lighting/lighting_source.dm

```diff
@@ -74,6 +74,35 @@
 
 	return ..()
 
+///add this light source to new_atom_host's light_sources list. updating movement registrations as needed
+/datum/light_source/proc/add_to_light_sources(atom/new_atom_host)
+	if(QDELETED(new_atom_host))
+		return FALSE
```

Comment:
do we need this check?

Reply:
yes unless youd rather it error i suppose